lib/js/components/DatePickers/DatePicker/DatePicker.vue

```diff
@@ -0,0 +1,278 @@
+<template>
+	<div
+		class="ds-datePicker"
+		:class="{ '-ds-box': triggerType === DATE_PICKER_TRIGGER_TYPES.BOX }"
+	>
+		<template v-if="triggerType === DATE_PICKER_TRIGGER_TYPES.TILE">
+			<span
+				class="ds-datePicker__label"
+				:class="{
+					'-ds-isUppercase': isLabelUppercase,
+					'-ds-disabled': state === DATE_PICKER_STATES.DISABLED,
+				}"
+				>{{ label }}</span
+			>
+			<ds-tile
+				:text="text"
+				:interactive="isInteractive"
+				:color="color"
+				:state="state"
+				:icon-right="icon"
+				:is-icon-right-hidden-on-mobile="isIconHiddenOnMobile"
+				:eyebrow-text="eyebrowText"
+				@click="toggle"
+			/>
+		</template>
+		<date-picker-box
+			v-else
+			:is-interactive="isInteractive"
+			:placeholder="placeholder"
+			:date="date"
+			:icon="icon"
+			:is-icon-hidden-on-mobile="isIconHiddenOnMobile"
+			:state="state"
+			:color="color"
+			:eyebrow-text="eyebrowText"
+			:is-open="isOpen"
+			@click="toggle"
+		/>
+
+		<span class="ds-datePicker__errorMessage"> {{ errorMessage }} </span>
+		<input ref="flatpickrInput" class="ds-datePicker__hiddenInput" />
+	</div>
+</template>
+
+<style lang="scss">
+@import '../../../../styles/settings/typography/tokens';
+@import '../../../../styles/settings/colors/tokens';
+@import '../../../../styles/settings/spacings';
+@import 'flatpickr/dist/flatpickr';
+
+.ds-datePicker {
+	$self: &;
+
+	display: flex;
+	flex-direction: column;
+	row-gap: $space-4xs;
+
+	&.-ds-box {
+		display: inline-flex;
+	}
+
+	&__hiddenInput {
+		display: none;
+	}
+
+	&__label {
+		@include label-l-default-bold;
+
+		color: $color-neutral-text-strong;
+		padding-bottom: $space-4xs;
+
+		&.-ds-isUppercase {
+			@include label-m-extensive-bold-uppercase;
+
+			color: $color-neutral-text;
+		}
+
+		&.-ds-disabled {
+			color: $color-neutral-text-strong-disabled;
+
+			&.-ds-isUppercase {
+				color: $color-neutral-text-disabled;
+			}
+		}
+	}
+
+	&__errorMessage {
+		@include info-s-default-regular;
+
+		color: $color-danger-text;
+	}
+}
+</style>
+
+<script lang="ts">
+import { toRaw, PropType } from 'vue';
+
+import DsTile from '../../Tile';
+import { ICONS } from '../../Icons/Icon';
+import DatePickerBox from '../DatePickerBox';
+
+import {
+	DATE_PICKER_CALENDAR_POSITIONS,
+	DATE_PICKER_COLORS,
+	DATE_PICKER_STATES,
+	DATE_PICKER_TRIGGER_TYPES,
+	DatePickerCalendarPositions,
+	DatePickerColors,
+	DatePickerStates,
+	DatePickerTriggerTypes,
+} from './DatePicker.consts';
+import { capitalizeFirstLetter } from '../../../../../tools/importers/helpers/modifiers';
+
+let flatpickr: Function | null = null;
+
+const flatpickrPositions = {
```

Comment:
```suggestion
const FLATPICKR_POSITIONS = {
```
for consistency with consts